plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -238,10 +220,8 @@ func (e *Elasticsearch) gatherNodeStats(url string, acc telegraf.Accumulator) er
 			"cluster_name": nodeStats.ClusterName,
 		}
 
-		if e.ClusterStats {
-			// check for master
-			e.isMaster = (id == e.catMasterResponseTokens[0])
-		}
+		// check for master
+		e.localNodeIsMaster = (id == e.masterNodeId)
```

Comment:
Sorry for the delay in response. My thought is that the master-only statistics related to this flag should be restricted to only work when local=true. Then only one node would be reported in the node-stats, and this flag would make sense in that context.  This is actually the way we're configuring local here, which seems to work fine. If memory serves, I removed that restriction due to a comment against an earlier version. I'm thinking I should put that restriction back in. Would this approach work for you?